prpl/attachments.c

```diff
@@ -192,3 +194,432 @@ void download_attachment(ChimeConnection *cxn, ChimeAttachment *att, AttachmentC
 	data->ctx = ctx;
 	purple_util_fetch_url_len(att->url, TRUE, NULL, FALSE, ATTACHMENT_MAX_SIZE, download_callback, data);
 }
+
+/*
+ * Chime Attachment Upload
+ *
+ * Uploading a file through Chime involves many steps.
+ * This is basically the currently flow:
+ *
+ * +--------+         +--------+     +----+   +---------+
+ * | Chime  |         | Chime  |     | S3 |   |Recipient|
```

Comment:
from this comment I see that you interact with S3 though I do not see the code where you are supposed to sign the message in order to be able to interact with the s3 rest api... is that not really needed for this use case?

Reply:
All the parameters necessary for S3 comes embedded on the url returned by the Chime server. All we need to do is a `PUT` to this url.

Reply:
ah! ok, I guess it would be nice to have this as a comment on the code

Reply:
Sure, done!